src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -68,16 +68,25 @@ GcsServer::GcsServer(const ray::gcs::GcsServerConfig &config,
           PeriodicalRunner::Create(io_context_provider_.GetDefaultIOContext())),
       is_started_(false),
       is_stopped_(false) {
-  // Init GCS table storage.
+  // Init GCS table storage. Note this is on the default io context, not the one with
+  // GcsInternalKVManager, to avoid congestion on the latter.
   RAY_LOG(INFO) << "GCS storage type is " << storage_type_;
+  auto &io_context = io_context_provider_.GetDefaultIOContext();
   switch (storage_type_) {
   case StorageType::IN_MEMORY:
-    gcs_table_storage_ = std::make_unique<InMemoryGcsTableStorage>(
-        io_context_provider_.GetDefaultIOContext());
+    gcs_table_storage_ = std::make_unique<InMemoryGcsTableStorage>(io_context);
     break;
-  case StorageType::REDIS_PERSIST:
-    gcs_table_storage_ = std::make_unique<gcs::RedisGcsTableStorage>(GetOrConnectRedis());
+  case StorageType::REDIS_PERSIST: {
```

Comment:
just a question, not request to change, redis doesn't have durability guarantee (even if we config primary/secondary and enable periodical persistence), why do we call it `PERSIST`?

Reply:
this meant if the redis is alive, and the gcs is dead and restarted, it can read back all info from redis. What naming do you suggest to change to?

Reply:
I just feel weird on the wording "persist" since redis doesn't provide durability guarantee. I'm thinking just call it redis

Reply:
We can save the discussion to a separate PR. It may be more than just a renaming.